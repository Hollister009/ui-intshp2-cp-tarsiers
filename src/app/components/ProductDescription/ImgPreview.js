/* eslint-disable max-len */
/* eslint-disable react/prefer-stateless-function */
/* eslint-disable jsx-a11y/alt-text */
import React, { Component } from 'react';
// import ImgSwitch from './ImgSwitch'
// import ActiveThunb from './ActiveTumb';
import './Test.scss';

export default class ImgPreview extends Component {
  constructor(props) {
    super(props);
    const { item } = this.props;

    this.state = {
      figureStyle: {
        backgroundImage: `url(${item.src})`,
        backgroundPosition: '0% 0%'
      },
      mainImgURL: item.src
    };
  }

  handleMouseMove = e => {
    const { left, top, width, height } = e.target.getBoundingClientRect();
    const x = ((e.pageX - left) / width) * 100;
    const y = ((e.pageY - top) / height) * 100;

    this.setState(state => ({
      ...state,
      figureStyle: { ...state.figureStyle, backgroundPosition: `${x}% ${y}%` }
    }));
  };

  changeMainImage = e => {
    const newMainUrl = e.target.getAttribute('src');

    this.setState({
      mainImgURL: newMainUrl,
      figureStyle: { backgroundImage: `url(${newMainUrl})` }
    });
  };

  render() {
    const { item } = this.props;
    const { figureStyle } = this.state;
    const thumbnails = Object.values(item.colorUrls).map(el => (
      <img src={el} alt="" key={el} onClick={this.changeMainImage} />
    ));

    return (
      <div className="img-container">
        <figure onMouseMove={this.handleMouseMove}>
          <img src={this.state.mainImgURL} />
          <div style={figureStyle} />
        </figure>
        <div className="thumbnails-box">{thumbnails}</div>
      </div>
    );
  }
}
