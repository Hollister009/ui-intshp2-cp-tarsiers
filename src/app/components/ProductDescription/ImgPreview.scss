@import '../../../styles/mixin';
@import '../../../styles/variables';

.imgpreview {
  border: 1px solid black;
}
