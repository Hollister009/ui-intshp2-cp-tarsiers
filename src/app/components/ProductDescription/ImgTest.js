/* eslint-disable max-len */
/* eslint-disable react/prefer-stateless-function */
/* eslint-disable jsx-a11y/alt-text */
import React, { Component } from 'react';
import './Test.scss';

const src =
  'https://res.cloudinary.com/so/image/upload/v1548069002/products/reebok%20jacket/red.jpg';

export default class ImgTest extends Component {
  constructor() {
    super();
    this.state = {
      figureStyle: {
        backgroundImage: `url(${src})`,
        backgroundPosition: '0% 0%'
      }
    };
  }

  handleMouseMove = e => {
    const { left, top, width, height } = e.target.getBoundingClientRect();
    const x = ((e.pageX - left) / width) * 100;
    const y = ((e.pageY - top) / height) * 100;

    this.setState(state => ({
      ...state,
      figureStyle: {
        backgroundImage: `url(${src})`,
        backgroundPosition: `${x}% ${y}%`
      }
    }));
  };

  render() {
    const { figureStyle } = this.state;

    return (
      <div className="test-container">
        <figure onMouseMove={this.handleMouseMove} style={figureStyle}>
          <img src={src} />
        </figure>
      </div>
    );
  }
}
