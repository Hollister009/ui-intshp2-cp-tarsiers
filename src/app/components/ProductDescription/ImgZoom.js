/* eslint-disable max-len */
/* eslint-disable react/prefer-stateless-function */
/* eslint-disable jsx-a11y/alt-text */
import React, { Component } from 'react';
import './Test.scss';

// const src =
//   'https://res.cloudinary.com/so/image/upload/v1548069002/products/reebok%20jacket/red.jpg';

export default class ImgZoom extends Component {
  constructor(props) {
    super(props);

    const { item } = this.props;

    this.state = {
      figureStyle: {
        backgroundImage: `url(${item.src})`,
        backgroundPosition: '0% 0%'
      }
    };
  }

  handleMouseMove = e => {
    const { left, top, width, height } = e.target.getBoundingClientRect();
    const x = ((e.pageX - left) / width) * 100;
    const y = ((e.pageY - top) / height) * 100;

    this.setState(state => ({
      ...state,
      figureStyle: {
        backgroundImage: `url(${item.src})`,
        backgroundPosition: `${x}% ${y}%`
      }
    }));
  };

  render() {
    const { figureStyle } = this.state;

    return <div onMouseMove={this.handleMouseMove} style={figureStyle} />;
  }
}
