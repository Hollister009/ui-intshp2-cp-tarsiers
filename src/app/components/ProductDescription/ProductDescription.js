import React, { Component } from 'react';
import appConfig from '../../../config/appConfig';

import styles from './ProductDescription.module.scss';

class ProductDescription extends Component {
  constructor(props) {
    super(props);
    this.state = { quantity: 0 };
  }

  increment = () => {
    this.setState(prevState => ({ quantity: prevState.quantity + 1 }));
  };

  decrement = () => {
    const { quantity } = this.state;

    if (quantity > 0) {
      this.setState(prevState => ({ quantity: prevState.quantity - 1 }));
    }
  };

  render() {
    const { item } = this.props;
    const { quantity } = this.state;

    if (!item) {
      return null;
    }
    const price = quantity > 0 ? item.price * quantity : item.price;
    const sizes = item.sizes.map((element, index, array) => (
      <React.Fragment>
        <a href="/" className={styles.size} onClick={e => e.preventDefault()}>
          {element}
        </a>
        {index + 1 !== array.length ? <span>-</span> : null}
      </React.Fragment>
    ));

    return (
      <section className={styles.section}>
        <div className={styles.preview} />
        <div className={styles.details}>
          <div className={styles.info}>
            <h1 className={styles.title}>{item.title}</h1>
            <h3 className={styles.subtitle}>
              {appConfig.productDescription.subheader}
            </h3>
            <p className={styles.text}>
              {appConfig.productDescription.description}
            </p>
          </div>
          <div className={styles.select}>
            <div className={styles.flex_row}>
              <p className={styles.choose}>Choose Size</p>
              <div className={styles.sizes}>{sizes}</div>
            </div>
            <div className={styles.flex_row}>
              <p className={styles.choose}>Choose Quantity</p>
              <div className={styles.quantity}>
                <button type="button" onClick={this.increment}>
                  &#43;
                </button>
                <span>{quantity}</span>
                <button type="button" onClick={this.decrement}>
                  &#45;
                </button>
              </div>
            </div>
          </div>
          <div className={styles.order}>
            <p className={styles.price}>{`Price: ${price}$`}</p>
            <div className={styles.buttons_row}>
              <button type="button">
                <i className="fas fa-globe" />
              </button>
              <button type="button">
                <i className="fas fa-cart-plus" />
              </button>
              <button type="button">
                <i className="far fa-heart" />
              </button>
              <button type="button" className={styles.btn_order}>
                Order Now
              </button>
            </div>
          </div>
        </div>
      </section>
    );
  }
}

export default ProductDescription;
