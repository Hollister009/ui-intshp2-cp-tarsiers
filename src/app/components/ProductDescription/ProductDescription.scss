@import '../../../styles/mixin';
@import '../../../styles/variables';

p {
  font-size: 18px;
  font-family: $main-font-regular;
  font-weight: 400;
}

.product-details {
  @include d-flex(center);
}

.images {
  width: 70%;
}

.product-description {
  @include d-flex(inherit, column);

  margin: 100px 40px 0 40px;
}

.product-text {
  @include d-flex(inherit, column);

  .product-header {
    color: $color-main;
    font-family: $main-font-bold;
    font-size: 52px;
    text-transform: uppercase;
    line-height: 45px;
  }

  .product-subheader {
    color: $color-dark;
    font-family: $main-font-regular;
    font-weight: 400;
    font-size: 18px;
    margin: 30px 0 15px 0;
  }

  .product-info {
    color: $color-dark;
    font-family: $main-font-regular;
    font-size: 16px;
    line-height: 30px;
    margin: 15px 0 30px 0;
  }
}

.size-quantity {
  @include d-flex();

  border-top: 2px solid $color-light-2;
  border-bottom: 2px solid $color-light-2;
  padding: 30px 0;

  .choose {
    margin-right: 20px;
  }

  .p-size {
    @include d-flex();

    .sizes {
      @include d-flex();

      .size {
        color: $color-dark;
        font-size: 18px;
        text-decoration: none;
        padding: 0 5px;
        text-transform: uppercase;

        &:hover {
          animation: 0.5s scale forwards;
          color: $color-main;

          @keyframes scale {
            from {
              transform: scale(1);
            }
            to {
              transform: scale(1.1);
            }
          }
        }

        &:focus-within {
          color: $color-main;
          font-size: 18px;
        }
      }
    }
  }

  .p-quantity {
    @include d-flex();

    .quantity {
      @include d-flex();

      align-items: flex-start;

      button {
        background: none;
        font-size: 18px;
        padding: 0 10px;
      }

      span {
        color: $color-main;
      }
    }
  }
}

.order-section {
  @include d-flex();

  padding: 30px 0;
  align-items: center;

  .price {
    font-weight: bold;
  }

  .order-buttons {
    @include d-flex();

    align-items: center;

    button {
      background: none;
      padding: 0 10px;
      color: $color-dark;
      font-size: 18px;

      &:hover {
        color: $color-main;
      }
    }

    .order-now {
      color: $color-white;
      background-color: $color-main;
      border-radius: 30px;
      padding: 20px 30px;
      font-size: 16px;

      &:hover {
        background-color: $color-dark;
      }
    }
  }
}
