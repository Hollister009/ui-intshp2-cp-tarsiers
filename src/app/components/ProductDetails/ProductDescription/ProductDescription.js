import React, { Component } from 'react';
import { Flags } from 'react-feature-flags';
import { Notify } from 'react-redux-notify';
import PropTypes from 'prop-types';

import { productType } from '../../../types';
import appConfig from '../../../../config/appConfig';
import NotifyService from '../../../../utils/notify.service';
import { addItem, removeItem } from '../../../../utils/wishlist.service';

import styles from './ProductDescription.module.scss';

class ProductDescription extends Component {
  static propTypes = {
    item: productType,
    wished: PropTypes.bool,
    inCart: PropTypes.bool
  };

  static defaultProps = { item: null, wished: false, inCart: false };

  state = { quantity: 1, sizeClicked: '', activeColor: '' };

  addItem = addItem.bind(this);

  removeItem = removeItem.bind(this);

  increment = () => {
    this.setState(prevState => ({ quantity: prevState.quantity + 1 }));
  };

  decrement = () => {
    const { quantity } = this.state;

    if (quantity > 1) {
      this.setState(prevState => ({ quantity: prevState.quantity - 1 }));
    }
  };

  toggleSizes = e => {
    e.preventDefault();

    this.setState({ sizeClicked: e.target.innerText.toLowerCase() });
  };

  toggleColors = e => {
    e.preventDefault();

    this.setState({ activeColor: e.target.innerText });
  };

  toggleWishList = (e, id) => {
    const { wished } = this.props;

    e.preventDefault();

    const cb = !wished ? this.addItem : this.removeItem;

    cb(id);
  };

  toggleCart = (e, id) => {
    const {
      inCart,
      removeFromCart,
      addToCart,
      createNotification
    } = this.props;

    e.preventDefault();

    const cb = !inCart ? addToCart : removeFromCart;

    if (!inCart) {
      createNotification(NotifyService.cartAdd);
    } else {
      createNotification(NotifyService.cartRemove);
    }

    cb(id);
  };

  orderHandler = e => {
    e.preventDefault();
    const { item, orderNowItem, createNotification } = this.props;
    const { sizeClicked, quantity, activeColor } = this.state;
    const orderData = {
      title: item.title,
      size: sizeClicked,
      color: activeColor,
      price: item.price,
      src: item.src,
      quantity
    };

    if (sizeClicked && activeColor) {
      orderNowItem(orderData);
      createNotification(NotifyService.ordered);
    } else if (sizeClicked) {
      createNotification(NotifyService.chooseColor);
    } else {
      createNotification(NotifyService.chooseSize);
    }
  };

  render() {
    const { item, wished, inCart, children } = this.props;
    const { quantity, sizeClicked } = this.state;

    if (!item) {
      return null;
    }

    const { _id } = item;
    const price = quantity > 1 ? item.price * quantity : item.price;
    const sizes = item.sizes.map((element, index, array) => {
      const active = sizeClicked === element ? { color: '#ff5912' } : {};

      return (
        <React.Fragment key={element}>
          <a
            href="/"
            className={styles.size}
            onClick={e => this.toggleSizes(e)}
            style={active}
          >
            {element}
          </a>
          {index + 1 !== array.length ? <span>-</span> : null}
        </React.Fragment>
      );
    });
    const swatches = item.colors.map(color => {
      const { activeColor } = this.state;
      const style = { backgroundColor: `${color}` };

      return (
        <span
          key={color}
          role="button"
          tabIndex="0"
          className={color === activeColor ? styles.active : styles.color}
          style={style}
          onClick={e => this.toggleColors(e)}
          onKeyDown={e => this.toggleColors(e)}
        >
          {color}
        </span>
      );
    });

    return (
      <div className={styles.background}>
        <div className="container">
          <h1 className={styles.titlebig} id="test">
            {item.title}
          </h1>
          <section className={styles.section}>
            <div className={styles.preview}>{children}</div>
            <div className={styles.details}>
              <div className={styles.info}>
                <h1 className={styles.title}>{item.title}</h1>
                <h3 className={styles.subtitle}>
                  {appConfig.productDescription.subheader}
                </h3>
                <p className={styles.text}>
                  {appConfig.productDescription.description}
                </p>
              </div>
              <div className={styles.select}>
                <div className={styles.flex_row}>
                  <p className={styles.choose}>Choose Size</p>
                  <div className={styles.sizes}>{sizes}</div>
                </div>
                <div className={styles.flex_row}>
                  <p className={styles.choose}>Choose Color</p>
                  <div className={styles.colors}>{swatches}</div>
                </div>
                <div className={styles.flex_row}>
                  <p className={styles.choose}>Choose Quantity</p>
                  <div className={styles.quantity}>
                    <button
                      type="button"
                      onClick={this.increment}
                      data-type="increment"
                    >
                      &#43;
                    </button>
                    <span>{quantity}</span>
                    <button
                      type="button"
                      onClick={this.decrement}
                      data-type="decrement"
                    >
                      &#45;
                    </button>
                  </div>
                </div>
              </div>
              <div className={styles.order}>
                <p className={styles.price}>{`Price: ${price}$`}</p>
                <div className={styles.buttons_row}>
                  <button type="button">
                    <i className="fas fa-globe" />
                  </button>
                  <Notify position={NotifyService.position.topRight} />
                  <button
                    type="button"
                    title="Add to shopping-cart"
                    onClick={e => this.toggleCart(e, _id)}
                    data-type="cart-btn"
                  >
                    {inCart ? (
                      <i className="fas fa-cart-arrow-down highlighted" />
                    ) : (
                      <i className="fas fa-cart-plus" />
                    )}
                  </button>
                  <Flags authorizedFlags={[appConfig.killswitch.wishlist]}>
                    <button
                      type="button"
                      onClick={e => this.toggleWishList(e, _id)}
                      title="Add to wish-list"
                      data-type="wishlist-btn"
                    >
                      <i
                        className={
                          wished ? 'fas fa-heart highlighted' : 'fas fa-heart'
                        }
                      />
                    </button>
                  </Flags>
                  <button
                    type="button"
                    className={styles.btn_order}
                    onClick={e => this.orderHandler(e)}
                  >
                    Order Now
                  </button>
                </div>
              </div>
            </div>
          </section>
        </div>
      </div>
    );
  }
}

export default ProductDescription;
