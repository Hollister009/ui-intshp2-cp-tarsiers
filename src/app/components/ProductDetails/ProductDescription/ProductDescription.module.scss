@import '../../../../styles/mixin';
@import '../../../../styles/variables';

.background {
  background-color: $color-white;
  border-top: 0.5px solid $color-light-2;
  padding-bottom: 65px;

  .titlebig {
    display: none;
    color: $color-main;
    font-family: $main-font-bold;
    font-size: 52px;
    text-transform: uppercase;
    line-height: 45px;
    margin-top: 30px;
    text-align: center;
    margin-bottom: 20px;
  }
}

.active {
  border: 2px solid $color-main;
}

.section {
  @include d-flex(center);

  p {
    font-size: 18px;
    font-family: $main-font-regular;
    font-weight: 400;
  }

  .flex_row {
    @include d-flex();
  }
  .choose {
    margin-right: 20px;
  }
  .price {
    font-weight: bold;
  }
}

.preview {
  width: 60%;
}

.details {
  margin: 0;
  margin-top: 100px;

  .info {
    @include d-flex(flex-start, column);

    & > .title {
      color: $color-main;
      font-family: $main-font-bold;
      font-size: 52px;
      text-transform: uppercase;
      line-height: 45px;
    }
    & > .subtitle {
      color: $color-dark;
      font-family: $main-font-regular;
      font-weight: 400;
      font-size: 18px;
      margin: 30px 0 15px 0;
    }
    & > .text {
      color: $color-dark;
      font-family: $main-font-regular;
      font-size: 16px;
      line-height: 30px;
      margin: 15px 0 30px 0;
    }
  }

  .select {
    padding: 30px 0;
    border-top: 2px solid $color-light-2;
    border-bottom: 2px solid $color-light-2;
    @include d-flex();

    .sizes {
      @include d-flex() {
        align-items: center;
      }
    }

    .colors {
      @include d-flex() {
        align-items: center;
      }

      span {
        font-size: 0px;
        display: block;
        width: 15px;
        height: 15px;
        border-radius: 50%;
        margin: 2px;
        margin-bottom: 5px;
        cursor: pointer;
        outline: none;

        &:hover {
          animation: scale 0.5s forwards ease;
        }
      }
    }

    .size {
      color: $color-dark;
      font-size: 18px;
      padding: 0 5px;
      text-decoration: none;
      text-transform: uppercase;
      outline: none;

      &:hover {
        animation: scale 0.5s forwards ease;
        color: $color-main;
      }
    }

    .activate {
      color: $color-main;
      font-size: 22px;
      padding: 0 5px;
      text-decoration: none;
      text-transform: uppercase;
      outline: none;
    }

    .quantity {
      @include d-flex() {
        align-items: baseline;
      }

      button {
        background: none;
        font-size: 18px;
        padding: 0;

        &:first-of-type {
          padding-left: 10px;
        }

        &:last-of-type {
          padding-right: 10px;
        }
      }

      span {
        color: $color-main;
        font-size: 18px;
        padding: 0 10px;
      }
    }
  }

  .order {
    padding: 30px 0;
    @include d-flex();

    & > .buttons_row {
      @include d-flex();

      button {
        background: none;
        padding: 0 10px;
        color: $color-dark;
        font-size: 18px;

        &:hover {
          color: $color-main;
        }
      }

      .btn_order {
        color: $color-white;
        background-color: $color-main;
        border-radius: 30px;
        padding: 20px 30px;
        font-size: 16px;

        &:hover {
          background-color: $color-dark;
        }
      }
    }
  }
}

@keyframes scale {
  from {
    transform: scale(1);
  }
  to {
    transform: scale(1.1);
  }
}

@media screen and (max-width: $media-tablet) {
  .background {
    .titlebig {
      display: flex;
      flex-direction: row;
      justify-content: center;
    }
  }
  .section {
    @include d-flex(center, column);
  }

  .details {
    margin-top: 20px;
  }

  .preview {
    width: 100%;
  }

  .title {
    display: none;
  }
}

@media screen and (max-width: $media-mobile) {
  .section {
    @include d-flex(center, column);
  }

  .preview {
    width: 100%;
  }

  .details {
    .select {
      @include d-flex(unset, column);

      .flex_row {
        align-items: center;
      }
    }

    .order {
      @include d-flex(unset, column);

      align-items: center;

      .price {
        padding: 25px 0;
      }
    }
  }
}

@media screen and (max-width: $media-smallest) {
  .section {
    @include d-flex(center, column);
  }

  .preview {
    width: 100%;
  }

  .title {
    text-align: center;
    font-size: 40px !important;
  }

  .details {
    .select {
      @include d-flex(unset, column);

      .flex_row {
        align-items: center;
      }
    }

    .order {
      @include d-flex(unset, column);

      align-items: center;

      .price {
        padding: 25px 0;
      }
    }
  }
}
