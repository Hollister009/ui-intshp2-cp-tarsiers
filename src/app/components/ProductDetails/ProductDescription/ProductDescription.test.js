import React from 'react';
import { shallow } from 'enzyme';
import ProductDescription from './ProductDescription';

const products = require('../../../../mocks/products.json');

describe('<ProductDescription />', () => {
  let props;

  beforeEach(() => {
    props = {
      item: products[2],
      wished: false,
      inCart: false,
      addToCart: jest.fn(),
      removeFromCart: jest.fn(),
      orderNowItem: jest.fn(),
      createNotification: jest.fn()
    };
  });

  it('should match its snapshot', () => {
    const wrapper = shallow(<ProductDescription {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should return null when item is falsy', () => {
    const wrapper = shallow(<ProductDescription />);

    expect(wrapper.html()).toBe(null);
  });

  it('should be able to increment', () => {
    const wrapper = shallow(<ProductDescription {...props} />);
    const { quantity } = wrapper.state();

    wrapper.find('button[data-type="increment"]').simulate('click');
    expect(wrapper.state().quantity).toBe(quantity + 1);
  });

  it('should be able to decrement', () => {
    const wrapper = shallow(<ProductDescription {...props} />);

    wrapper.setState({ quantity: 1 });
    wrapper.update();

    const decBtn = wrapper.find('button[data-type="decrement"]');

    decBtn.simulate('click');
    expect(wrapper.state().quantity).toBe(0);
    decBtn.simulate('click');
    expect(wrapper.state().quantity).toBe(0);
  });

  it('should be able to call toggleSizes', () => {
    const wrapper = shallow(<ProductDescription {...props} />);
    const spy = jest.spyOn(wrapper.instance(), 'toggleSizes');
    const size = wrapper.find('.size').at(0);
    const evt = { preventDefault() {}, target: { size, innerText: 'S' } };

    expect(wrapper.state().sizeClicked).toBe('');
    size.simulate('click', evt);

    expect(wrapper.state().sizeClicked).toBe('s');
    expect(spy).toHaveBeenCalled();
  });

  it('should be able to call orderHandler', () => {
    const evt = { preventDefault() {} };
    const wrapper = shallow(<ProductDescription {...props} />);
    const btn = wrapper.find('.btn_order');

    btn.simulate('click', evt);
    expect(props.orderNowItem).toHaveBeenCalled();
  });

  describe('toggleWishList', () => {
    it('should call addItem when wished is false', () => {
      const evt = { preventDefault() {} };
      const wrapper = shallow(<ProductDescription {...props} />);
      const spy = jest.spyOn(wrapper.instance(), 'addItem');
      const btn = wrapper.find('button[data-type="wishlist-btn"]');

      btn.simulate('click', evt);
      expect(wrapper.state().heartDisabled).toBe(true);
      expect(spy).toHaveBeenCalled();
    });

    it('should call removeItem when wished is true', () => {
      const evt = { preventDefault() {} };
      const wrapper = shallow(<ProductDescription {...props} />);
      const spy = jest.spyOn(wrapper.instance(), 'removeItem');
      const btn = wrapper.find('button[data-type="wishlist-btn"]');

      wrapper.setProps({ wished: true });
      wrapper.update();

      btn.simulate('click', evt);
      expect(wrapper.state().heartDisabled).toBe(false);
      expect(spy).toHaveBeenCalled();
    });
  });

  describe('toggleCart', () => {
    it('should call addToCart when inCart is false', () => {
      const id = props.item._id;
      const evt = { preventDefault() {} };
      const wrapper = shallow(<ProductDescription {...props} />);
      const btn = wrapper.find('button[data-type="cart-btn"]');

      btn.simulate('click', evt);
      expect(props.addToCart).toHaveBeenCalledWith(id);
    });

    it('should call removeFromCart when inCart is true', () => {
      const id = props.item._id;
      const evt = { preventDefault() {} };
      const wrapper = shallow(<ProductDescription {...props} />);
      const btn = wrapper.find('button[data-type="cart-btn"]');

      wrapper.setProps({ inCart: true });
      wrapper.update();

      btn.simulate('click', evt);
      expect(props.removeFromCart).toHaveBeenCalledWith(id);
    });
  });
});
