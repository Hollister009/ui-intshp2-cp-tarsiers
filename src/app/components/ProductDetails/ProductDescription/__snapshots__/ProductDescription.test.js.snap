// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDescription /> should match its snapshot 1`] = `
<div
  className="background"
>
  <div
    className="container"
  >
    <h1
      className="titlebig"
      id="test"
    >
      Nike Sports Suit
    </h1>
    <section
      className="section"
    >
      <div
        className="preview"
      />
      <div
        className="details"
      >
        <div
          className="info"
        >
          <h1
            className="title"
          >
            Nike Sports Suit
          </h1>
          <h3
            className="subtitle"
          >
            Lorem Ipsum is simply dummy text
          </h3>
          <p
            className="text"
          >
            Lorem Ipsum is simply dummy text of the printing and typesetting industry Lorem Ipsum is simply dummy text of the printing and typesetting industry
          </p>
        </div>
        <div
          className="select"
        >
          <div
            className="flex_row"
          >
            <p
              className="choose"
            >
              Choose Size
            </p>
            <div
              className="sizes"
            >
              <a
                className="size"
                href="/"
                onClick={[Function]}
                style={Object {}}
              >
                s
              </a>
              <span>
                -
              </span>
              <a
                className="size"
                href="/"
                onClick={[Function]}
                style={Object {}}
              >
                m
              </a>
              <span>
                -
              </span>
              <a
                className="size"
                href="/"
                onClick={[Function]}
                style={Object {}}
              >
                l
              </a>
              <span>
                -
              </span>
              <a
                className="size"
                href="/"
                onClick={[Function]}
                style={Object {}}
              >
                xl
              </a>
            </div>
          </div>
          <div
            className="flex_row"
          >
            <p
              className="choose"
            >
              Choose Color
            </p>
            <div
              className="colors"
            >
              <span
                className="color"
                key="#171717"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                style={
                  Object {
                    "backgroundColor": "#171717",
                  }
                }
                tabIndex="0"
              >
                #171717
              </span>
            </div>
          </div>
          <div
            className="flex_row"
          >
            <p
              className="choose"
            >
              Choose Quantity
            </p>
            <div
              className="quantity"
            >
              <button
                data-type="increment"
                onClick={[Function]}
                type="button"
              >
                +
              </button>
              <span>
                1
              </span>
              <button
                data-type="decrement"
                onClick={[Function]}
                type="button"
              >
                -
              </button>
            </div>
          </div>
        </div>
        <div
          className="order"
        >
          <p
            className="price"
          >
            Price: 800$
          </p>
          <div
            className="buttons_row"
          >
            <button
              type="button"
            >
              <i
                className="fas fa-globe"
              />
            </button>
            <Connect(Notify)
              position="TopRight"
            />
            <button
              data-type="cart-btn"
              onClick={[Function]}
              title="Add to shopping-cart"
              type="button"
            >
              <i
                className="fas fa-cart-plus"
              />
            </button>
            <Flags
              authorizedFlags={
                Array [
                  "showWishlist",
                ]
              }
              exactFlags={false}
              renderOff={[Function]}
              renderOn={[Function]}
            >
              <button
                data-type="wishlist-btn"
                onClick={[Function]}
                title="Add to wish-list"
                type="button"
              >
                <i
                  className="fas fa-heart"
                />
              </button>
            </Flags>
            <button
              className="btn_order"
              onClick={[Function]}
              type="button"
            >
              Order Now
            </button>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
